cg/meta/orders/external_data.py

```diff
@@ -149,33 +148,93 @@ def create_dict(self, name: str) -> dict:
                         lims_sample_id,
                     )
                     return
                 last_version = self.housekeeper_api.last_version(bundle=lims_sample_id)
+
                 if not last_version:
                     LOG.info("Creating bundle for sample %s in housekeeper", lims_sample_id)
                     hk_dict: dict = self.create_dict(name=lims_sample_id)
                     bundle_result: Tuple[Bundle, Version] = self.create_hk_bundle(
                         bundle_name=lims_sample_id, data_dict=hk_dict, dry_run=dry_run
                     )
                     last_version = bundle_result[1]
-                    self.housekeeper_api.add_commit(bundle_result[0], bundle_result[1])
+                    files = []
+                else:
+                    files = [
+                        tmp.path
+                        for tmp in self.housekeeper_api.get_files(
+                            bundle=lims_sample_id, tags=["fastq"], version=last_version.id
+                        )
+                    ]
+
                 for path in paths:
+                    if path in files:
+                        LOG.info(
+                            "Path %s is already linked to bundle %s in housekeeper"
+                            % (path, lims_sample_id)
+                        )
+                        continue
+
                     LOG.info("Adding path %s to bundle %s in housekeeper" % (path, lims_sample_id))
                     self.housekeeper_api.add_file(
                         path=path, version_obj=last_version, tags=["fastq"]
                     )
-                    self.housekeeper_api.commit()
+                    do_commit = True
+                    if os.path.isfile(path + ".md5"):
+                        do_commit = check_md5sum(fastq_path=path)
+                    else:
+                        LOG.info("No md5 file found for file  %s" % path + ".md5")
+                    if not do_commit:
+                        return
+        self.housekeeper_api.commit()
+        self.set_to_analyze(cases=cases)
+
+    def set_to_analyze(self, cases):
+        for case_obj in cases:
+            case_obj.action = "analyze"
+            self.status_db.commit()
+        LOG.info("Setting all cases to analyze.")
+
+
+def check_md5sum(fastq_path) -> bool:
+    """Checks if there is and md5 file associated with the fastq-file and if so verifies the cheksum"""
+    with open(fastq_path, "rb") as f:
+        file_hash = hashlib.md5()
+        chunk = f.read(8192)
+        while chunk:
+            file_hash.update(chunk)
+            chunk = f.read(8192)
+    outp_sum = file_hash.hexdigest()
+    with open(fastq_path + ".md5", "r") as md:
+        check_sum = extract_md5sum(md.readline())
+    if check_sum == outp_sum:
+        LOG.info("The md5 file matches the md5sum for file  %s" % fastq_path)
+        return True
+    else:
```

Comment:
codefactor here sais you dont need "else", just unindent and return

Reply:
Removed redundant else